src/wizard/wizard-form.tsx

```diff
@@ -42,6 +43,13 @@ export default function WizardFormWithAnalytics(props: WizardFormProps) {
     props.steps[props.activeStepIndex] as BasePropsWithAnalyticsMetadata
   );
   const __internalRootRef = useComponentMetadata('WizardForm', PACKAGE_VERSION, { ...analyticsMetadata });
+  const stepHeaderRef = useRef<HTMLDivElement | null>(null);
+
+  useEffectOnUpdate(() => {
```

Comment:
Needed to hoist the effect up a level for it to work. When used directly inside the WizardForm it got a new reference each time and would always think its the first render. Might be something we need to look more into with the AnalyticsFunnelStep and what its doing to cause this.